pyccel/codegen/printing/cwrappercode.py

```diff
@@ -782,11 +792,14 @@ def _print_FunctionDef(self, expr):
         # Save all used names
         used_names = set([a.name for a in expr.arguments] + [r.name for r in expr.results] + [expr.name.name])
 
+        # update ndarray local variables properties
+        local_arg_vars = [a.clone(a.name, is_pointer=True) if isinstance(a, Variable) and a.rank > 0 else a for a in expr.arguments]
```

Comment:
will the variable `a` have `allocatable` == `true` and `is_pointer` == `true` ?

Reply:
yes, you are right, it should not have allocatable and is_pointer both True.
I'll fix it.